autoload/coqtail.vim

```diff
@@ -406,6 +406,10 @@ function! coqtail#start(...) abort
       \ 'deprecated': has('python')})
     call s:call('refresh', '', 0, {})
 
+    if coqtail#util#getvar([b:, g:], 'coqtail_proof_diffs', 0)
```

Comment:
Another issue I just thought of here is that this will not work for versions of Coq before proof diffs were available.

How should this version checking happen on the vim side? Or should it be delegated to the Python side?

Reply:
Looks like proof diffs are supported [since version 8.9.0](https://coq.github.io/doc/master/refman/changes.html#id744).

I adapted your code that checks for Coq version support, to determine when to query with `Set Diffs "{arg}"` at start (called in `coqtail#start()`):

```
function! s:setup_proofdiffs(coqtail_version) abort
  let l:proof_diffs_arg = coqtail#util#getvar([b:, g:], 'coqtail_auto_enable_proof_diffs', "")
  if l:proof_diffs_arg == ""
    return
  endif

  let l:versions = map(split(a:coqtail_version, '\v(\.|pl)'), 'str2nr(v:val)')
  " Pad missing version numbers with 0
  while len(l:versions) < 3
    let l:versions = add(l:versions, 0)
  endwhile

  for l:supp in s:proof_diffs_supported
    let l:is_sup = 1
    for l:idx in range(3)
      if l:supp[l:idx] != l:versions[l:idx] && l:supp[l:idx] != -1
        let l:is_sup = 0
        break
      endif
    endfor
    if l:is_sup
      call s:call("query", "", 0, {"args": ["Set", "Diffs", '"' . l:proof_diffs_arg . '"' ]})
      return
    endif
  endfor
endfunction
```

Is there a better way to do this? Seems a bit clunky for just one feature..

Reply:
Also, for what it's worth, I tried using the `"removed"` option. I'm not really sure what Coqtop is trying to do based on what Coqtail is showing, but it makes it really hard to tell what's going on.

Moving between a context of:

```
1 subgoal
C : Prop
T : Set
α : T -> Prop
H : exists x : T, C -> α x
c : C
========================= (1 / 1)
exists x : T, α x
```

and 

```
1 subgoal
C : Prop
T : Set
α : T -> Prop
x : T
H : C -> α x
c : C
========================= (1 / 1)
exists x0 : T, α x0
```

I get this:

![image](https://user-images.githubusercontent.com/11800204/103447750-61b73c00-4c5d-11eb-8af7-8f69942e4a6d.png)

Reply:
I just merged #178, which adds a `coqtail#version#atleast` function that you can use to check if `b:coqtail_version` is 8.9.0 or greater.